private-click-measurement.bs

```diff
@@ -25,120 +25,179 @@ Complain About: missing-example-ids yes
-    - That it will want ad click attribution for this click.
-    - What the intended [=ad click destination=] is.
-    - What the [=ad campaign id=] is.
-4. The user agent navigates and takes note that the user landed on the intended [=ad click destination=].
-5. The user's activity on the [=ad click destination=] leads to a [=conversion=].
-6. A third-party HTTP request is made on the ad click destination website to `​https://search.example/.well-known/ad-click-attribution`
-7. The user agent checks for pending ad click attributions for the [=ad click source-destination pair=] and if there's a hit, makes or schedules an HTTP request to `​https://search.example/.well-known/ad-click-attribution` with the [=ad attribution data=].
+3. The [=click source website=], `search.example`, informs the user agent (see [[#linkformat]]):
+    - That it will want attribution for this click.
+    - What the intended [=attribute-on website=] is.
+    - What the [=source id=] is.
+4. The user agent navigates and takes note that the user landed on the intended [=attribute-on website=].
+5. The user's activity on the [=attribute-on website=] leads to a [=triggering event=].
+6. A third-party HTTP request is made on the [=click source website=] to `​https://search.example/.well-known/private-click-measurement/trigger-attribution/` with [=trigger data=].
+7. The user agent checks for stored clicks for the [=click source-attribute-on pair=] and if there's a hit, makes or schedules an HTTP POST request to `​https://search.example/.well-known/private-click-measurement/report-attribution/` with the corresponding [=attribution report=].
 
-    ISSUE: One thing to consider here is whether there should be an option to send the [=ad attribution data=] to the [=ad click destination=] too.
+    ISSUE: One thing to consider here is whether there should be an option to send the [=attribution report=] to the [=attribute-on website=] too.
 
-# Ad Click Source Link Format # {#linkformat}
+# Click Source Link Format # {#linkformat}
 
-This specification adds two attributes to the {{HTMLAnchorElement}} interface. Authors can use these attributes in HTML content like so (where `17` is an [=ad campaign id=] and `https://destination.example/` is an [=ad click destination=]):
+This specification adds two attributes to the {{HTMLAnchorElement}} interface. Authors can use these attributes in HTML content like so (where `17` is an [=source id=] and `https://destination.example/` is an [=attribute-on website=]):
 
 <xmp class="highlight" highlight=html>
-<a adcampaignid="17" addestination="https://destination.example/">
+<a attributionsourceid="17" attributeon="https://destination.example/">
 </xmp>
 
 Formally:
 
 <pre class="idl">
 partial interface HTMLAnchorElement {
-    [CEReactions] attribute DOMString adCampaignId;
-    [CEReactions] attribute USVString adDestination;
+    [CEReactions] attribute DOMString attributionSourceId;
```

Comment:
Why is this not an integer?

Reply:
Most IDs can contain (almost) arbitrary characters; I assume that's the case here?

Reply:
No: https://privacycg.github.io/private-click-measurement/#ad-campaign-id

Reply:
Oh, yeah, then this should definitely be an integer (probably unsigned long to take best advantage of HTML's reflection abilities).

Reply:
I can’t find where but the reason I believe was layering. Google wants to be able to have far more entropy and might want to use more characters than digits. There was also talk of hierarchical IDs such as “<low entropy value>:<high entropy value>” that can either allow developers to have one markup that supports both or allow the browser to use one of the two or both depending on user settings or some privacy measurement.

WebKit has had this as an experimental feature since May 2019 and is five weeks into beta for official release. Changing it will likely be a breaking change for us.

Reply:
Sounds like unsigned long then. Hopefully, we won't have too many who've adopted PCM in WebKit browsers by the time we can get this change out.

Reply:
@johnwilander for most uses of this API it might not even be a breaking change as JavaScript is pretty lenient.

Reply:
Looks like 8623b5d addresses this concern, yeah?